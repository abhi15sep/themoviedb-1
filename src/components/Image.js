import React from 'react';
import PropTypes from 'prop-types';
import CONSTANTS from '../constants';

const { HOST, DEFAULT, SIZE } = CONSTANTS.IMAGE;

function Image (props) {
  const { size, path, title } = props;
  const src = path ?
              `${HOST}${size}${path}` :
              (size === SIZE.SMALL) ?
              DEFAULT.SMALL : DEFAULT.LARGE;

  return (
    <img src={src} alt={title} title={title}/>
  );
}

Image.propTypes = {
  size: PropTypes.string.isRequired,
  path: PropTypes.string,
  title: PropTypes.string.isRequired,
}

export default Image;
