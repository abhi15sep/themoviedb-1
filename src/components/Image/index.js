import React from 'react';
import PropTypes from 'prop-types';
import CONSTANTS from '../../utils/constants';

const { HOST } = CONSTANTS.IMAGE;

const getNoPosterSrc = size => size.NO_POSTER;

const getDefaultSrc = (size, path) => `${HOST}${size.DEFAULT}${path}`;

const Image = (props) => {
  const { size, path, title } = props;
  const src = path ? getDefaultSrc(size, path) : getNoPosterSrc(size);

  return <img src={src} alt={title} title={title} />;
};

Image.propTypes = {
  size: PropTypes.shape({
    DEFAULT: PropTypes.string.isRequired,
    NO_POSTER: PropTypes.string.isRequired,
  }).isRequired,
  path: PropTypes.string,
  title: PropTypes.string.isRequired,
};

Image.defaultProps = {
  path: '',
};

export default Image;
