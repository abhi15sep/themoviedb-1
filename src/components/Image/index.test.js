import React from 'react';
import { render } from '@testing-library/react';
import Image from './index';

const title = 'Underwater';
const path = '/gzlbb3yeVISpQ3REd3Ga1scWGTU.jpg';
const size = {
  DEFAULT: 'w300',
  NO_POSTER: 'https://dummyimage.com/300x450/eeeeee/999999&text=No+poster',
};

test('renders image', () => {
  const { getByAltText } = render(<Image title={title} size={size} path={path} />);
  const image = getByAltText('Poster of Underwater');

  expect(image).toBeInTheDocument();
  expect(image.src).toBe('https://image.tmdb.org/t/p/w300/gzlbb3yeVISpQ3REd3Ga1scWGTU.jpg');
});

test('renders no-poster image', () => {
  const { getByAltText } = render(<Image title={title} size={size} />);
  const image = getByAltText('Poster of Underwater');

  expect(image).toBeInTheDocument();
  expect(image.src).toBe('https://dummyimage.com/300x450/eeeeee/999999&text=No+poster');
});
