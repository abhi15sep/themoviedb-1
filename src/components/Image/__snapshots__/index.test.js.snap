// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  alt="The Movie"
  src="http://image.tmdb.org/t/p/w185/path/to/image"
  title="The Movie"
/>
`;

exports[`renders with no poster image 1`] = `
<img
  alt="The Movie"
  src="/path/to/no-poster-image"
  title="The Movie"
/>
`;
